File: app/demo/src/main/java/net/gotev/recycleradapterdemo/EmptyItem.java
package net.gotev.recycleradapterdemo;

import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import net.gotev.recycleradapter.AdapterItem;
import net.gotev.recycleradapter.RecyclerAdapterNotifier;

import butterknife.BindView;

/**
 * @author Aleksandar Gotev
 */

public class EmptyItem extends AdapterItem<EmptyItem.Holder> {

    private String text;

    public EmptyItem(String text) {
        this.text = text;
    }

    @Override
    public void onItemChanged(Bundle dataChanged) {

    }

    @Override
    public int getLayoutId() {
        return R.layout.item_empty;
    }

    @Override
    protected void bind(EmptyItem.Holder holder) {
        holder.textView.setText(text);
    }

    public static class Holder extends ButterKnifeViewHolder {

        @BindView(R.id.textView)
        TextView textView;

        public Holder(View itemView, RecyclerAdapterNotifier adapter) {
            super(itemView, adapter);
        }
    }
}

File: app/demo/src/main/java/net/gotev/recycleradapterdemo/adapteritems/EmptyItem.kt
package net.gotev.recycleradapterdemo.adapteritems

import android.view.View
import android.widget.TextView
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.item_empty.*

import net.gotev.recycleradapter.AdapterItem
import net.gotev.recycleradapter.RecyclerAdapterNotifier

import net.gotev.recycleradapter.RecyclerAdapterViewHolder
import net.gotev.recycleradapterdemo.R


class EmptyItem(private val text: String) : AdapterItem<EmptyItem.Holder>() {

    override fun getLayoutId() = R.layout.item_empty

    override fun bind(holder: Holder) {
        holder.textViewField.text = text
    }

    class Holder(itemView: View, adapter: RecyclerAdapterNotifier) : RecyclerAdapterViewHolder(itemView, adapter), LayoutContainer {

        override val containerView: View?
            get() = itemView

        internal val textViewField: TextView by lazy { textView }
    }
}
